.github/workflows/ci.yaml

```diff
@@ -29,26 +29,43 @@ jobs:
       - name: Swift tests
         run: swift test
 
-  build-wasm32-unknown-wasi:
+  wasm:
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        include:
+          - toolchain: swift-DEVELOPMENT-SNAPSHOT-2024-10-08-a
+            swift-sdk: swift-wasm-DEVELOPMENT-SNAPSHOT-2024-10-15-a
+            checksum: e27d5df268cc9ecab2f5bca8ca693c8b684015de60593e1d472a78ce28ed9283
     steps:
       - uses: actions/checkout@v4
+      - uses: bytecodealliance/actions/wasmtime/setup@v1
       - uses: actions-rs/toolchain@v1
         with:
           profile: minimal
-          toolchain: 1.78.0
+          toolchain: 1.80.0
           default: true
-      - run: rustup target add wasm32-wasi
-      - uses: swiftwasm/setup-swiftwasm@v1
-        with:
-          swift-version: "wasm-5.9.1-RELEASE"
-      #- run: cargo build --manifest-path rust/Cargo.toml --target wasm32-wasi --release
+      - run: rustup target add wasm32-wasip1-threads
```

Comment:
Providing context: `wasm32-wasi` has been renamed to `wasm32-wasip1` [May'24], and it will be deprecated next January [Jan'25] [[ref](https://blog.rust-lang.org/2024/04/09/updates-to-rusts-wasi-targets.html#renaming-wasm32-wasi-to-wasm32-wasip1)]. `wasm32-wasip1-threads` is an extension of `wasm32-wasip1` adding native multi-threading capabilities. Mainly it supports std fully, [bulk memory](https://github.com/WebAssembly/spec/blob/main/proposals/bulk-memory-operations/Overview.md), atomics, expose pthread APIs. Both are Tiers2 in Rust, and for `wasm32-wasip1-threads` it will use atomic operations for sharable resources.

Said that we can not mix objects compiled between `wasm32-wasip1-threads` and `wasm32-wasip1`; so, I think it makes sense to generate a `libuniffi_automerge` with both, and let the users pick the correct one.